File: app/src/main/java/com/aiprotection/NotificationListener.java
package com.aiprotection;

import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;

public class NotificationListener extends NotificationListenerService {
    
    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        String packageName = sbn.getPackageName();
        String title = sbn.getNotification().extras.getString(android.app.Notification.EXTRA_TITLE, "");
        String text = sbn.getNotification().extras.getString(android.app.Notification.EXTRA_TEXT, "");
        
        String message = title + " " + text;
        
        // Простая проверка на опасные слова
        String[] dangerousWords = {"код из смс", "выиграл", "приз", "скинь деньги", "аккаунт взломан"};
        
        boolean isDangerous = false;
        for (String word : dangerousWords) {
            if (message.toLowerCase().contains(word)) {
                isDangerous = true;
                break;
            }
        }
        
        if (isDangerous) {
            Log.w("AI Protection", "⚠️ ОПАСНОЕ СООБЩЕНИЕ: " + message);
        }
    }
    
    @Override
    public void onNotificationRemoved(StatusBarNotification sbn) {}
}

File: app/src/main/java/com/aiprotection/NotificationListener.kt
package com.aiprotection

import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification
import android.util.Log
import okhttp3.*
import org.json.JSONObject
import java.io.IOException

class NotificationListener : NotificationListenerService() {
    
    private val client = OkHttpClient()
    
    override fun onNotificationPosted(sbn: StatusBarNotification?) {
        sbn?.let {
            val packageName = it.packageName
            val title = it.notification?.extras?.getString(android.app.Notification.EXTRA_TITLE) ?: ""
            val text = it.notification?.extras?.getString(android.app.Notification.EXTRA_TEXT) ?: ""
            val fullMessage = "$title $text"
            
            when {
                packageName.contains("telegram") -> checkWithAI(fullMessage, "TELEGRAM")
                packageName.contains("vk") -> checkWithAI(fullMessage, "VK")
            }
        }
    }
    
    private fun checkWithAI(message: String, messenger: String) {
        val dangerousWords = listOf(
            "код из смс", "выиграл", "приз", "скинь деньги", 
            "аккаунт взломан", "карта заблокирована", "госуслуги",
            "подтверди вход", "перейди по ссылке", "ваучер"
        )
        
        for (word in dangerousWords) {
            if (message.lowercase().contains(word)) {
                Log.w("AI Protection", "⚠️ ОПАСНО в $messenger: $message")
                break
            }
        }
    }
    
    override fun onNotificationRemoved(sbn: StatusBarNotification?) {}
}
